testsuite/testcases/src/dag_onchain_enable_test.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::NetworkLoadTest;
+use anyhow::Ok;
+use aptos::test::CliTestFramework;
+use aptos_forge::{NetworkTest, NodeExt, SwarmExt, Test};
+use aptos_logger::info;
+use aptos_sdk::bcs;
+use aptos_types::{
+    account_config::CORE_CODE_ADDRESS,
+    on_chain_config::{DagConsensusConfigV1, OnChainConsensusConfig},
+};
+use std::{fmt::Write, time::Duration};
+use tokio::runtime::Runtime;
+
+const MAX_NODE_LAG_SECS: u64 = 360;
+
+pub struct DagOnChainEnableTest {}
+
+impl Test for DagOnChainEnableTest {
+    fn name(&self) -> &'static str {
+        "dag reconfig enable test"
+    }
+}
+
+impl NetworkLoadTest for DagOnChainEnableTest {
+    fn test(
+        &self,
+        swarm: &mut dyn aptos_forge::Swarm,
+        _report: &mut aptos_forge::TestReport,
+        duration: std::time::Duration,
+    ) -> anyhow::Result<()> {
+        let runtime = Runtime::new().unwrap();
+
+        let faucet_endpoint: reqwest::Url = "http://localhost:8081".parse().unwrap();
+        let rest_client = swarm.validators().next().unwrap().rest_client();
+
+        let mut cli = runtime.block_on(async {
+            CliTestFramework::new(
+                swarm.validators().next().unwrap().rest_api_endpoint(),
+                faucet_endpoint,
+                /*num_cli_accounts=*/ 0,
+            )
+            .await
+        });
+
+        std::thread::sleep(duration / 2);
```

Comment:
why?

Reply:
we let the current consensus run with load for some time before we flip the flag.